rfcs/0010-multi-tenant-workload-identity/README.md

```diff
@@ -0,0 +1,1268 @@
+respective cloud resources. This field would be optional, and when not present the
+original behavior would be observed, i.e. the feature only activates
+when the field is present and a cloud provider among `aws`, `azure` or `gcp` is
+specified in the `spec.provider` field (or similar, see the last paragraph of this
+section). If only the `spec.provider` field is present, then the controller would
+use single-tenant workload identity, i.e. the identity of the controller itself.
+
+Alongside `spec.serviceAccountName` in the Flux APIs, we propose adding the binary
+flag `--use-service-account-for-provider-auth`. By default this flag would be set
+to `false`, meaning that the controller would use single-tenant workload identity
+for all the objects, i.e. the identity of the controller itself. When set to `true`,
+workload identity would require the `spec.serviceAccountName` field to be present
+in the object, and the controller would use the identity of the `ServiceAccount`
+specified in the object for accessing the cloud provider resources.
+
+In addition to this field, we propose adding the field `spec.sts.endpoint` for
+specifying the STS endpoint to use for exchanging the Kubernetes token for the
+cloud provider access token. This field would be optional, and when not present
+the behavior would be using the default STS endpoint for the cloud provider.
+At the time of writing this RFC, not all cloud providers support custom STS
+endpoints, so this field would be implemented only for the cloud providers
+that support it. This field is notably already present in the `Bucket` API,
+and is used for both EKS single-tenant workload identity in the `aws` bucket
+provider, and for LDAP authentication in the `generic` bucket provider.
+
+In the specific case of the `Bucket` API the `spec.sts` field has
+additional use cases. It has the additional field `spec.sts.provider`.
+When `spec.sts`, which is optional, is specified, then both
+`spec.sts.provider` and `spec.sts.endpoint` must be specified.
+This is not necessary for the other Flux APIs, so we propose
+only the `spec.sts.endpoint` field at the moment. The field `spec.sts`
+would be optional and `spec.sts.endpoint` would be mandatory when
+`spec.sts` is specified.
+
+When talking to a Security Token Service for exchanging tokens it's possible
+to use an HTTP/S proxy. Most Flux APIs already support the field
+`spec.proxySecretRef` for accessing the respective cloud resources, and some
+already use it for talking to the STS service in the single-tenant version
+of workload identity as well, e.g. `Bucket`. We propose using this field for
+specifying the proxy to use when talking to the STS service also in the
+multi-tenant version of workload identity.
+
+Finally, the `Kustomization` API uses Key Management Services (KMS) for
+decrypting SOPS-encrypted secrets. We propose adding the optional field
+`spec.decryption.key` with the required sub-field `.provider`: One of
```

Comment:
This does not match the Kustomization API from use-cases where we use `spec.decryption.keyProvider`. I think `keyProvider` is better than having `spec.decryption.key.provider`.

Reply:
Fixed, thanks!